### ISSUE TITLE
cross compile from linux to macos failed because of libpq.so, libsqlite3.so....

### ISSUE THREAD

**Issue by GopherJ** [diesel-rs/diesel#2510]
### Versions

- **Rust:** rustc 1.40.0 (73528e339 2019-12-16)
- **Diesel:** latest
- **Database:** postgres
- **Operating System** linux mint 18.04

### Feature Flags

- **diesel:** default

### What are you trying to accomplish?

cross compile diesel from linux to macos

### What is the expected output?
success

### What is the actual output?

fail

### Steps to reproduce

I'm basically following https://wapl.es/rust/2019/02/17/rust-cross-compile-linux-to-macos.html

1. install deps:
```
sudo apt install \
    clang \
    gcc \
    g++ \
    zlib1g-dev \
    libmpc-dev \
    libmpfr-dev \
    libgmp-dev
```

2. add apple target

```
rustup target add x86_64-apple-darwin
```

3. clone diesel
```
git clone https://github.com/diesel-rs/diesel
```

4. clone osxcross
```
cd diesel
git clone https://github.com/tpoechtrager/osxcross
cd osxcross
wget -nc https://s3.dockerproject.org/darwin/v2/MacOSX10.10.sdk.tar.xz
mv MacOSX10.10.sdk.tar.xz tarballs/
UNATTENDED=yes OSX_VERSION_MIN=10.7 ./build.sh

cd ../ # make sure we are in diesel root directory
```

5. configure cargo

```
mkdir .cargo
cat > .cargo/config <<EOF
[target.x86_64-apple-darwin]
linker = "x86_64-apple-darwin14-clang"
ar = "x86_64-apple-darwin14-ar"
EOF
```

6. cross compile diesel

```
PATH="$(pwd)/osxcross/target/bin:$PATH" \
CC=o64-clang \
CXX=o64-clang++ \
 PKG_CONFIG_ALLOW_CROSS=1 \
LIBZ_SYS_STATIC=1 \
cargo build --target x86_64-apple-darwin
```

7. got errors:

```
  = note: ld: warning: ignoring file /usr/lib/x86_64-linux-gnu/libsqlite3.so, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x7F 0x45 0x4C 0x46 0x02 0x01 0x01 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 )
          ld: warning: ignoring file /usr/lib/x86_64-linux-gnu/libpq.so, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x7F 0x45 0x4C 0x46 0x02 0x01 0x01 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 )
          ld: warning: ignoring file /usr/lib/x86_64-linux-gnu/libmysqlclient.so, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x7F 0x45 0x4C 0x46 0x02 0x01 0x01 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 )
          ld: warning: ignoring file /usr/lib/x86_64-linux-gnu/libc.so, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x2F 0x2A 0x20 0x47 0x4E 0x55 0x20 0x6C 0x64 0x20 0x73 0x63 0x72 0x69 0x70 0x74 )
          ld: warning: ignoring file /usr/lib/x86_64-linux-gnu/libresolv.so, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x7F 0x45 0x4C 0x46 0x02 0x01 0x01 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 )
          ld: warning: ignoring file /usr/lib/x86_64-linux-gnu/libm.so, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x2F 0x2A 0x20 0x47 0x4E 0x55 0x20 0x6C 0x64 0x20 0x73 0x63 0x72 0x69 0x70 0x74 )

```

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)

**Comment by GopherJ** [comment]
Have any one tried to cross compile diesel?

**Comment by weiznich** [comment]
Duplicate of #2352

**Comment by GopherJ** [comment]
@weiznich I know it's duplicated, but is there a potential solution for this? diesel is not async and is now not able to be cross compiled, what shall we do? supporting sqlx-rs and write sql manually?

**Comment by weiznich** [comment]
@GopherJ I consider open knowingly duplicate issues as offensive action against the maintainers of this crate. It's rude to just request that something will get magically solved that's in the end just a problem with your environment (as already written in the duplicated issue). In the future: please spend some time on reading and understanding error messages and already existing bug reports on similar issues before starting opening new duplicated issues. Any future duplicated issue from you will be closed without comment.

> what shall we do? supporting sqlx-rs and writting sql manually?

That's entirely up to you to decide. Nobody is forcing you to use diesel.
